File: C++/Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        
        int longest_run = 0;
        bool run = true;
        std::string current = strs[0];
        // strs.erase(strs.begin()); //* Clear it out.

        while (longest_run < current.length() && run) {

            for (int i = 1; i < strs.size(); i++) {
                if (strs[i].length() < longest_run || current[longest_run] != strs[i][longest_run]) {
                    run = false;
                    break;
                }
            }

            if (run)
                longest_run++;

        }
        
        current.resize(longest_run);
        return current;

    }
};

File: python/Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        longest_run = 0
        run = True
        # Pick the 1st item and match with all other items.

        c = strs[0]

        while longest_run < len(c) and run:

            for x in strs[1:]:
                if longest_run >= len(x):
                    run = False
                    break

                if c[longest_run] != x[longest_run]:
                    run = False
                    break
                    # Even if one doesn't match, break out.
            
            if run:
                longest_run += 1
            
        return c[0:longest_run]
